build/scripts/meta.yaml.schema

```diff
@@ -219,9 +219,38 @@
                         "type": "string",
                         "pattern": "^[-._a-zA-Z0-9]+$"
                       }
-                    },
-                    "mountSources": {
-                      "type": "boolean"
+                    }
+                  }
+                }
+              },
+              "mountSources": {
+                "type": "boolean"
+              }
+            },
+            "lifecycle": {
```

Comment:
I'm not sure `lifecycle` is correctly placed here -- shouldn't it be in `.containers.items[].properties`?

Reply:
AFAIR it's just the same as K8s container is... And properties - is just map[string,string], it would mean that we would have two properties `postStartCommand`, `preStopCommand` and then we'd transform it into k8s container...
I don't have a strong preference here.

@l0rd WDYT? Is it a good idea to make plugin model closer to k8s?

Reply:
@sleshchenko that's ok for this PR. We need to make sure version 2.0 of the devfile is compatible with it. In particular in version 2.0 we want to define plugins as usual devfile components (of type container/kubernetes/openshift) but defined in an external devfile. And we want that introduce a new devfile section called `events`. cc @davidfestal